### DISCUSSION TITLE
参数含义问题

### DISCUSSION THREAD

**Discussion by monkback** [isno/theByteBook#54] [General]
在2.2章中网络请求耗时计算中，有一些的参数解释似乎有点问题呢，像time_connect 就是指的tcp握手时间，而不是time_connect - tcp_namelookup；类似time_appconnect就是SSL握手耗时，而不是time_appconnect - time_connect。我也是参考着https://curl.se/docs/manpage.html 和 https://speedtestdemon.com/a-guide-to-curls-performance-metrics-how-to-analyze-a-speed-test-result/ 这两篇文章看的哈

**Comment by isno** [comment]
你列的文章写错了。

下面是 curl 里面计算时间的代码， t_startsingle 是请求开始时间（包含DNS解析）。

`case TIMER_CONNECT: data->progress.t_connect = Curl_tvdiff_secs(Curl_tvnow(), data->progress.t_startsingle);
`
所以，tcp 握手时间就是 time_connect - tcp_namelookup

我写的内容是正确的。
